includes/class-wpcom-legacy-redirector.php

```diff
@@ -227,13 +224,13 @@ public static function get_url_hash( $url ) {
 	 */
 	public static function normalise_url( $url ) {
 		// Sanitise the URL first rather than trying to normalise a non-URL.
-		$url = esc_url_raw( $url );
+		$url = esc_url_raw( wp_unslash( $url ) );
```

Comment:
I did add a new test under UtilsTest->test_normalise_url which should be able to clear this. 

So, the `wp_unslash` does not remove the extra slashes, but the redirection only works if the exact path is matched, so it needs to include any additional slashes, do we need to reconsider this?

Reply:
There are two points here - what the ideal situation would be, and what we currently need to maintain so that existing saved redirects continue to work. Changing how URLs are normalised worries me that we're going to break existing redirects.

Does the non-ascii redirects NEED this unslashing change?

Reply:
I've added the wp_unslash() for better sanitization, but it might as well be removed since we decompose the URL using `mb_parse_url` method.

I did add a few more tests to handle non-ASCII multiple slashes redirects, and I don't see it affecting the previous logic.

I think it's better to address your concern with a Functional test, maybe a CLI command that would run against the redirects found in the database, and then we apply the `Lookup::get_redirect_data` method to check the output, I think this will give us better confidence when dealing with a large number of redirects setup in knowing if we would break existing ones.

Reply:
removed the `wp_unslash` call for now.